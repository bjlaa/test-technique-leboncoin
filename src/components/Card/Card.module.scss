@import '../../variables';

.card {
  border-radius: 10px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  padding: 20px;
  margin: 20px auto;
  max-width: 80vw;
}

@media (min-width: $min-width-tablet) and (max-width: $min-width-laptop) {
  .card {
    max-width: 80vw;
    display: flex;
    flex-wrap: wrap;
  }
}

@media (min-width: $min-width-laptop) {
  .card {
    width: 50vw;
    max-width: 800px;
    display: flex;
    flex-wrap: wrap;
  }
}