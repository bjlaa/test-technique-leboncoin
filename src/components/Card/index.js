import React from 'react';
import PropTypes from 'prop-types';

import styles from './Card.module.scss';


const Card = ({ children }) => (
  <div className={styles.card}>
    {children}
  </div>
);

Card.propTypes = {
	children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

export default Card;