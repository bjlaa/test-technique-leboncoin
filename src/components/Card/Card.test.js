import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Card from './index';

configure({ adapter: new Adapter() });

describe('Card', () => {
  it('should match its snapshot', () => {
    const component = renderer.create(<Card />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render its children', () => {
    const child = <div id="child" />;

    const component = shallow(<Card children={child} />);

    expect(component.find('#child').length).toEqual(1);
  });
});
