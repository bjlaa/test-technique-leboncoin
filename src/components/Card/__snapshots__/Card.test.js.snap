// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match its snapshot 1`] = `
<div
  className="card"
/>
`;
